Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -110,18 +144,53 @@ public AndConstraint<TAssertions> BeOneOf(IEnumerable<string> validValues, strin
     /// <param name="becauseArgs">
     /// Zero or more objects to format using the placeholders in <paramref name="because" />.
     /// </param>
-    public AndConstraint<TAssertions> BeEquivalentTo(string expected, string because = "",
-        params object[] becauseArgs)
+    public AndConstraint<TAssertions> BeEquivalentTo(string expected,
+        string because = "", params object[] becauseArgs)
     {
         var expectation = new StringValidator(
-            new StringEqualityStrategy(StringComparison.OrdinalIgnoreCase),
+            new StringEqualityStrategy(StringComparer.OrdinalIgnoreCase),
             because, becauseArgs);
 
         expectation.Validate(Subject, expected);
 
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+    /// <summary>
+    /// Asserts that a string is exactly the same as another string, using the provided <paramref name="config"/>.
+    /// </summary>
+    /// <param name="expected">
+    /// The string that the subject is expected to be equivalent to.
+    /// </param>
+    /// <param name="config">
+    /// The equivalency options.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<TAssertions> BeEquivalentTo(string expected,
+        Func<EquivalencyAssertionOptions<string>, EquivalencyAssertionOptions<string>> config,
+        string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(config);
+
+        EquivalencyAssertionOptions<string> options = config(AssertionOptions.CloneDefaults<string>());
+
+        var expectation = new StringValidator(
+            new StringEqualityStrategy(options.GetStringComparerOrDefault()),
+            because, becauseArgs);
+
+        var subject = options.ApplyStringSettings(Subject);
+        expected = options.ApplyStringSettings(expected);
```

Comment:
Won't be indices in the failure message be wrong now?

Reply:
That's a very good question. I am unsure, what the expected behaviour should be, when I say "IgnoringLeadingWhitespace".
Do I expect the index to ignore it too or not?

I think you can argue both ways, so I would settle for the easier solution and leave it as it is.

But it would probably a good idea to explicitely mention it in the docs of `IgnoringLeadingWhitespace`...

Reply:
I mentioned the behaviour in the XML-Doc for `IgnoringLeadingWhitespace`.